src/kixi/collect/aggregate.clj

```diff
@@ -0,0 +1,41 @@
+(ns kixi.collect.aggregate
+  (:require [clojure.spec.alpha :as s]
+            [clojure.spec.gen.alpha :as gen]))
+
+(defprotocol IAggregateEventDataHandler
+  (get [this event])
+  (put! [this event output]))
+
+(defn satisfies-event-data-handler?
+  [handler]
+  (satisfies? IAggregateEventDataHandler handler))
+
+(s/fdef aggregate-event-handler-inner
+        :args (s/cat :create-handler-fn
+                     (s/fspec :args (s/cat :event :kixi/event)
+                              :ret (s/with-gen
+                                     satisfies-event-data-handler?
+                                     #(gen/return (reify IAggregateEventDataHandler
+                                                    (get [_ _])
+                                                    (put! [_ _ _])))))
+                     :process-event-fn
+                     (s/fspec :args (s/cat :data any?
+                                           :event :kixi/event)
+                              :ret any?)
+                     :event :kixi/event)
+        :ret nil?)
+
+(defn aggregate-event-handler-inner
+  [create-event-data-handler-fn process-event-fn event]
+  (let [handler (create-event-data-handler-fn event)]
+    (if (satisfies-event-data-handler? handler)
```

Comment:
I'm not sure you need this `if` check here now that you are checking for it in the fdef (and hopefully instrument or testing should show if you are using it wrong anywhere)

Reply:
Probably not, but this additional check just ensures that there's no run time funny business, or that an appropriate test hasn't been developed.